### PR TITLE
MGMT-10174: Add node ip hint manifest for sno.

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-10174](https://issues.redhat.com//browse/MGMT-10174): Add node ip hint manifest for sno.
We want to be able to set ip for kubelet that will match machine cidr
that we set for cluster. OVN will still run with default route one but
in sno case it doesn't influence cluster

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/cluster/cluster.go [hunk 1]
```diff
@@ -1268,6 +1268,11 @@ func (m *Manager) GenerateAdditionalManifests(ctx context.Context, cluster *comm
 		return errors.Wrap(err, "failed to add disk encryption manifest")
 	}
 
+	if err := m.manifestsGeneratorAPI.AddNodeIpHint(ctx, log, cluster); err != nil {
```
Comment (flaper87) [inline internal/cluster/cluster.go:1271]: should we call this only for `SNO` ? Or at least not fail in the `AddNodeIpHint` function if the machine cidr is not set. 

I believe this is useful only for SNO, tho.

Comment (tsorya) [reply internal/cluster/cluster.go:1271]: totally right, missed it. Changing to sno

Comment (tsorya) [reply internal/cluster/cluster.go:1271]: added inside function itself

File: internal/network/manifests_generator.go [hunk 2]
```diff
@@ -492,3 +494,94 @@ func NewConfig() (*Config, error) {
...
+	}
+	// if we have only one network there
+	// is not need to set those manifests
+	if len(hostNetworks) < 2 {
+		log.Debugf("SNO cluster has only one network, no need to add ip hint manifests")
+		return nil
+	}
+	log.Infof("Adding node add ip hint manifests")
+
+	for _, role := range []models.HostRole{models.HostRoleMaster, models.HostRoleWorker} {
```
Comment (eranco74) [inline internal/network/manifests_generator.go:555]: Why do we need the manifest for both roles?

Comment (tsorya) [reply internal/network/manifests_generator.go:555]: for day2

**Comment by tsorya** [comment]
/retitle Bug 2082979: Add node ip hint manifest for sno

**Pull request merged**